src/contracts/core/DelegationManager.sol

```diff
@@ -293,11 +310,24 @@ contract DelegationManager is Initializable, OwnableUpgradeable, Pausable, Deleg
         // retrieve `staker`'s list of strategies and the staker's shares in each strategy from the StrategyManager
         (IStrategy[] memory strategies, uint256[] memory shares) = strategyManager.getDeposits(staker);
 
-        // add strategy shares to delegated `operator`'s shares
+
+        //retrieve any beacon chain ETH shares the staker might have
+        uint256 beaconChainETHShares = eigenPodManager.podOwnerShares(staker);
+        IStrategy beaconChainETHStrategy = eigenPodManager.beaconChainETHStrategy();
+
         uint256 stratsLength = strategies.length;
+
+        if (beaconChainETHShares > 0){
```

Comment:
why involve the loop with this? just
```soldity
if (beaconChainETHShares > 0){
    operatorShares[operator][beaconChainETHStrategy] += beaconChainETHShares;
}
```

Reply:
fixed and made it work with the 'undelegation limbo' mechanism. good catch, not sure what sidu was thinking on this one https://github.com/Layr-Labs/eigenlayer-contracts/pull/135/commits/47d19c7b81ae980917026d7b3adaa95bd14cde39